### PR TITLE
Accountancy/feature/journal entries validation

### PR DESCRIPTION
**Pull request by Zank94**

### CODE REVIEW

File: app/controllers/backend/draft_journals_controller.rb [hunk 1]
```diff
@@ -39,7 +39,31 @@ class DraftJournalsController < Backend::BaseController
     end
 
     # this method lists all the entries generated in draft mode
-    def show; end
+    def show
+      @redirection = params[:redirection]
+      @current_page = 1
+      @current_date = Date.today
+      @journal_id = params[:journal_id] ? params[:journal_id].to_i : ''
+      journal_entries = @journal_id == '' ? JournalEntry.all : JournalEntry.where(journal_id: @journal_id)
```
Comment (DucMirack) [inline app/controllers/backend/draft_journals_controller.rb]: Replace `@journal_id == ''` by `@journal_id.blank?` for more readibility

File: app/controllers/backend/draft_journals_controller.rb [hunk 2]
```diff
@@ -39,7 +39,31 @@ class DraftJournalsController < Backend::BaseController
...
+      @draft_entries = @draft_entries.page(@current_page).per(20)
+      @unbalanced_entries_count = journal_entries.where('printed_on <= ?', @current_date).select { |entry| !entry.balanced? }.count
+      notify_warning_now(:there_are_x_remaining_unbalanced_entries, count: @unbalanced_entries_count) unless @unbalanced_entries_count < 1
+    end
+
+    def list
+      @redirection = params[:redirection]
+      @current_page = params[:page] ? params[:page].to_i : 1
+      @current_date = Date.parse(params[:to]) if params[:to]
+      @journal_id = params[:journal_id] == '' ? params[:journal_id] : params[:journal_id].to_i
```
Comment (DucMirack) [inline app/controllers/backend/draft_journals_controller.rb]: Replace `params[:journal_id] == ''` by `params[:journal_id].blank?` for more readibility ?
Is `params[:journal_id]` always set ?

File: app/controllers/backend/draft_journals_controller.rb [hunk 3]
```diff
@@ -39,7 +39,31 @@ class DraftJournalsController < Backend::BaseController
     end
 
     # this method lists all the entries generated in draft mode
-    def show; end
+    def show
+      @redirection = params[:redirection]
+      @current_page = 1
+      @current_date = Date.today
+      @journal_id = params[:journal_id] ? params[:journal_id].to_i : ''
```
Comment (DucMirack) [inline app/controllers/backend/draft_journals_controller.rb]: Is `params[:journal_id]` always set ?

File: app/controllers/backend/draft_journals_controller.rb [hunk 4]
```diff
@@ -65,5 +89,19 @@ def confirm
       end
       redirect_to action: :show
     end
+
+    def confirm_all
+      journal_id = params[:journal_id] == '' ? params[:journal_id] : params[:journal_id].to_i
+      journal_entries = journal_id == '' ? JournalEntry.all : JournalEntry.where(journal_id: journal_id)
```
Comment (DucMirack) [inline app/controllers/backend/draft_journals_controller.rb]: Replace  `journal_id == ''` by ` journal_id.blank?` for more readibility ?

File: config/locales/fra/action.yml [hunk 5]
```diff
@@ -1097,10 +1103,18 @@ fra:
...
     confirmed: "Confirmées" #?
     confirmed_at: "Confirmée le" #?
     connect: "Se connecter"
     consult: "Consulter" #?
+    consult_the_day_book:
+      zero: "Consulter le brouillard (%{count} écriture)"
+      one: "Consulter le brouillard (%{count} écriture)" #?
+      other: "Consulter le brouillard (%{count} écritures)" #?
+    consult_unbalanced_entries:
+      one: "Consulter les écritures à équilibrer (%{count})" #?
```
Comment (DucMirack) [inline config/locales/fra/action.yml]: Singular case : Change with "Consulter l'écriture ..."

File: config/locales/fra/action.yml [hunk 6]
```diff
@@ -1097,10 +1103,18 @@ fra:
       filter_on_attribute_of_class: "Filtre sur %{attribute} (%{class})"
       special-subscriber: "Filtre Abonné"
     configure: "Configurer"
+    confirmation: "Confirmation"
     confirmed: "Confirmées" #?
     confirmed_at: "Confirmée le" #?
     connect: "Se connecter"
     consult: "Consulter" #?
+    consult_the_day_book:
+      zero: "Consulter le brouillard (%{count} écriture)"
```
Comment (DucMirack) [inline config/locales/fra/action.yml]: You can consult day book if there is 0 journal entry ?

File: config/locales/fra/action.yml (hunk 5 above)
Comment (Zank94) [reply config/locales/fra/action.yml]: Following the design
screenshot-2018-7-31 accountancy 2018 mockups sales

Comment (DucMirack) [reply config/locales/fra/action.yml]: I was talking about line 1116 : `one: "Consulter les écritures à équilibrer (%{count})"`

Comment (Zank94) [reply config/locales/fra/action.yml]: My bad

**Pull request merged**